packages/editor/test/__snapshots__/ColorPicker.tsx.snap

```diff
@@ -4,7 +4,7 @@ exports[`renders with styles 1`] = `
 .emotion-9 {
   display: grid;
   padding: 8px;
-  grid-gap: 8px;
+  gap: 8px;
```

Comment:
Oh, this looks like unintended change. We'd like to preserve `gridGap` I think. Did you have any problem with the types with `gridGap`?

Reply:
Yeah I ran into a little rabbit hole with this one. 😄 It's intended, I had to change the [`gridGap`](https://github.com/system-ui/theme-ui/pull/900/files#diff-3a1a855ba5cdfdcc021fe760440538c7L86) property to [`gap`](https://github.com/system-ui/theme-ui/pull/900/files#diff-f952eb4f04acd9fa07eb7138709d1c4dR102) because TypeScript was complaining about the `gridGap` property not existing on the `SystemStyleObject` type.

That's because we define our CSS properties like this

```ts
export interface CSSProperties extends CSS.StandardProperties<number | string>, CSS.SvgProperties<number | string> {}
```

but only `gap` is part of `CSS.StandardProperties`. `gridGap` is part of `CSS.ObsoleteProperties` which are non-standard CSS props.

In this case `gridGap` is a deprecated prop and was changed to `gap` (so it can also be used for non-grid purposes). Technically `gap` might not work on some older versions of browsers which implemented the first version `gridGap`. But I'd think it would be more important to go forward with the standard.

What do you think? Should I rather include `CSS.ObsoleteProperties ` into the `CSSProperties` definition?

Reply:
Goddamn, TypeScript knows CSS better than I do...

That's a good question.

Should we support obsolete CSS properties on type level? I suppose that's an "responsible adult" thing to do, because more properties will possible become obsolete in the future and we'd rather avoid breaking people's builds because of this. We could implement a warning at runtime if we'd like to enforce the standard, but type error may be a bit too harsh, and less user friendly.

cc @jxnblk

---

BTW One can still use `sx={{ color: 'primary', ...{ gridGap: 2 } }}` as a workaround for the obsolete property.

Reply:
> I suppose that's an "responsible adult" thing to do, because more properties will possible become obsolete in the future and we'd rather avoid breaking people's builds because of this.

I agree! Should I open a PR to add `CSS.ObsoleteProperties`?